provider/provider.go

```diff
@@ -153,7 +153,15 @@ func NewWithHostDataSource(source io.Reader, options ...ProviderHandler) (*Wasmc
 	}
 
 	// Connect to NATS
-	nc, err := nats.Connect(hostData.LatticeRPCURL)
+	var nc *nats.Conn
+	if hostData.LatticeRPCUserSeed != "" && hostData.LatticeRPCUserJWT != "" {
+		opts := nats.UserJWTAndSeed(hostData.LatticeRPCUserJWT, hostData.LatticeRPCUserSeed)
+		logger.Debug("connecting to nats with userJWTAndSeed")
+		nc, err = nats.Connect(hostData.LatticeRPCURL, opts)
+	} else {
+		logger.Debug("connecting to nats", "url", hostData.LatticeRPCURL)
+		nc, err = nats.Connect(hostData.LatticeRPCURL)
+	}
```

Comment:
It might be preferable to connect once, optionally providing connection options if the appropriate configuration is provided, something like:
```go
	var ncopts []nats.Option
	if hostData.LatticeRPCUserSeed != "" && hostData.LatticeRPCUserJWT != "" {
		logger.Debug("Using UserJWT and Seed for connection")
		ncopts = append(ncopts, nats.UserJWTAndSeed(hostData.LatticeRPCUserJWT, hostData.LatticeRPCUserSeed))
	}
	nc, err = nats.Connect(hostData.LatticeRPCURL, ncopts)
```

Reply:
looks good, thanks!